src/Tracking/AdminActionHandler.php

```diff
@@ -54,9 +63,76 @@ public function optInToUsageTracking() {
 		}
 
 		give_update_option( AdminSettings::USAGE_TRACKING_OPTION_NAME, 'enabled' );
-		$this->usageTrackingOnBoarding->disableNotice( 'permanently' );
+		$this->usageTrackingOnBoarding->disableNotice( 0 );
+		$this->storeAccessToken();
 
 		wp_safe_redirect( remove_query_arg( 'give_action' ) );
 		exit();
 	}
+
+	/**
+	 * OptIn website to telemetry server when admin grant by changing setting.
+	 *
+	 * @since 2.10.0
+	 *
+	 * @param  array  $oldValue
+	 * @param  array  $newValue
+	 *
+	 * @return false
+	 */
+	public function optInToUsageTrackingAdminGrantManually( $oldValue, $newValue ) {
+		$class = __CLASS__;
+		add_filter( "give_disable_hook-update_option_give_settings:{$class}@optInToUsageTrackingAdminGrantManually", '__return_true' );
+
+		$section = isset( $_GET['section'] ) ? 'advanced-options' : '';
+		if ( ! Give_Admin_Settings::is_setting_page( 'advanced', $section ) ) {
+			return false;
+		}
+
+		$usageTracking = $newValue[ AdminSettings::USAGE_TRACKING_OPTION_NAME ] ?: 'disabled';
+		$usageTracking = give_is_setting_enabled( $usageTracking );
+
+		// Exit if already has access token.
+		if ( $usageTracking || get_option( OptionName::TELEMETRY_ACCESS_TOKEN ) ) {
+			return false;
+		}
+
+		$this->storeAccessToken();
+
+		remove_filter( "give_disable_hook-update_option_give_settings:{$class}@optInToUsageTrackingAdminGrantManually", '__return_false' );
+
+		return true;
+	}
+
+	/**
+	 * Store access token
+	 *
+	 * @since 2.10.0
+	 */
+	private function storeAccessToken() {
+		$client = new TrackClient();
+		$data   = array_merge(
+			( new ServerData() )->get(),
+			( new WebsiteData() )->get()
+		);
+
+		$response = $client->post( EventId::CREATE_TOKEN, $data, [ 'blocking' => true ] );
+		if ( is_wp_error( $response ) ) {
+			return;
+		}
+
+		$response = json_decode( wp_remote_retrieve_body( $response ), true );
+		if ( empty( $response['success'] ) ) {
+			return;
+		}
+
+		$token = $response['data']['access_token'];
+		update_option( OptionName::TELEMETRY_ACCESS_TOKEN, $token );
+
+		// Access token saved, now send first set of tracking information.
```

Comment:
@JasonTheAdams I notice that a few requests will be rejected by the telemetry server because of the rate limit set to `3`. Will, you suggest increasing the rate limit time to `10`?

Reply:
Yeah, that makes sense!

Reply:
fixed: https://github.com/impress-org/give-telemetry-server/compare/dda5b87f2e73...8abac5c31f72